src/NuGet.Clients/NuGet.CommandLine/Commands/Pack/AssemblyMetadataExtractor.cs

```diff
@@ -15,7 +15,7 @@ public static class AssemblyMetadataExtractor
     {
         private static T CreateInstance<T>(this AppDomain domain)
         {
-            return (T)domain.CreateInstanceAndUnwrap(typeof(T).Assembly.FullName, typeof(T).FullName);
+            return (T)domain.CreateInstanceFromAndUnwrap(Assembly.GetExecutingAssembly().Location, typeof(T).FullName);
```

Comment:
Given nuget.exe is ILMerged, it probably doesn't matter, but would keeping the `typeof(T)` be a bit safer, rather than `Assembly.GetExecutingAssembly()`?

Reply:
Technically yeah, if anyone ever moves these types around it would be safer, but there's a test to catch this. 

If you care I can change it but I'd reallly hate to try to fight with CI because of a change like that :D

lmk, this merge is not time critical so we can go through the CI churn if necessary.

Reply:
@zivkan Do you care about this or should I merge?